test/gurka/test_build_admin.cc

```diff
@@ -0,0 +1,281 @@
+#include <gtest/gtest.h>
+
+#include "baldr/admin.h"
+#include "filesystem.h"
+#include "gurka.h"
+#include "mjolnir/admin.h"
+#include "mjolnir/adminbuilder.h"
+#include "mjolnir/pbfadminparser.h"
+#include "mjolnir/pbfgraphparser.h"
+#include "test/test.h"
+
+using namespace valhalla;
+using namespace valhalla::baldr;
+using namespace valhalla::gurka;
+using namespace valhalla::mjolnir;
+
+namespace {
+
+// GetAdminInfo() requires a sqlite db handle and tiles. This creates a mock
+// graph with two states part of the same country - with a highway between them.
+valhalla::gurka::map BuildPBF(const std::string& workdir) {
+  const std::string ascii_map = R"(
+        A-------B-------C-------I-------J
+        |       |       |   W   |       |
+        |       |       |   |   |       |
+        |   G---|---H---|---X---|---Y   |
+        |       |       |   |   |       |
+        |       |       |   Z   |       |
+        F-------E-------D-------L-------K
+  )";
+
+  // To define an administrative boundary, the nodes must form a closed polygon.
+  const gurka::ways ways = {{"ABCDEFA", {}},
+                            {"ABEFA", {}},
+                            {"BCDEB", {}},
+                            {"CIJKLDC", {}},
+                            {"CILDC", {}},
+                            {"IJKLI", {}},
+                            {"GH",
+                             {
+                                 {"highway", "primary"},
+                             }},
+                            {"HX",
+                             {
+                                 {"highway", "primary"},
+                             }},
+                            {"XY",
+                             {
+                                 {"highway", "primary"},
+                             }},
+                            {"WX",
+                             {
+                                 {"highway", "primary"},
+                             }},
+                            {"XZ",
+                             {
+                                 {"highway", "primary"},
+                             }}};
+
+  // X lives Japan which allows named intersections - and is named.
+  const gurka::nodes nodes = {{"X", {{"junction", "named"}}}};
```

Comment:
i think it might be `junction=yes` and `name=Shibuya Crossing` something like that. im surprised that it sets the boolean even though it doesnt have a name!? that is blowing my mind. cc @gknisely

Reply:
https://wiki.openstreetmap.org/wiki/Tag:junction%3Dyes#Usage_examples

Reply:
Lol I dunno, I just reverse engineered this code:  https://github.com/valhalla/valhalla/blob/52885542d9bd5718d54f9ae8d01de072859cbab8/src/mjolnir/pbfgraphparser.cc#L872

Reply:
Sorry I don't know how to grab three lines in a link yet... its 872-874.

Reply:
no worries at all but i think you found another mild bug! notice the boolean `has_junction_name` regardless of whether or not it actually has a name? that doesnt seem right! and your test proves that the flag makes it all the way through to the nodeinfo after the tiles are created. i think we dont want to do that if there isnt actually a name. and oh man looking at that method it really needs the refactor stick. i'm going to push up a bit of a change to that part of the code, bear with me!